browser/brave_content_browser_client.cc

```diff
@@ -61,29 +61,52 @@ using extensions::ChromeContentBrowserClientExtensionsPart;
 
 namespace {
 
-bool HandleURLRewrite(GURL* url,
-                      content::BrowserContext* browser_context) {
-  if (url->SchemeIs(content::kChromeUIScheme) &&
-      (url->host() == chrome::kChromeUIWelcomeHost ||
-       url->host() == chrome::kChromeUIWelcomeWin10Host)) {
-    *url = GURL(kBraveUIWelcomeURL);
+bool HandleURLOverrideRewrite(GURL* url,
+                              content::BrowserContext* browser_context) {
+  // redirect sync-internals
+  if (url->host() == chrome::kChromeUISyncInternalsHost ||
+      url->host() == chrome::kChromeUISyncHost) {
+    GURL::Replacements replacements;
+    replacements.SetHostStr(chrome::kChromeUISyncHost);
+    *url = url->ReplaceComponents(replacements);
     return true;
   }
-  if (url->SchemeIs(content::kChromeUIScheme) &&
-      (url->host() == kBraveUISyncHost)) {
-    *url = GURL(kBraveUISyncURL);
+
+  // no special win10 welcome page
+  if (url->host() == chrome::kChromeUIWelcomeWin10Host ||
+      url->host() == chrome::kChromeUIWelcomeHost) {
+    *url = GURL(chrome::kChromeUIWelcomeURL);
```

Comment:
Q) Why do you make rest handler skip for these two hosts (sync and welcome) by returning true?
Do you think upstream url handler doesn't need to handle them?

Reply:
Ah! I saw upstream code tries to rewritten it. Ignore my above comment.